utils/ChromiumDownloader.js

```diff
@@ -217,3 +224,17 @@ function downloadFile(url, destinationPath, progressCallback) {
 function extractZip(zipPath, folderPath) {
   return new Promise(fulfill => extract(zipPath, {dir: folderPath}, fulfill));
 }
+
+function applyAgent(opt) {
+  const proxy_url = getProxyForUrl(opt.href);
+
+  if (proxy_url)
+  {
+    const parsedProxy = URL.parse(proxy_url);
+
+    if (parsedProxy.protocol === 'http:')
+      parsedProxy.secureProxy = false;
+
+    opt.agent = new ProxyAgent(parsedProxy);
```

Comment:
Constructing a result object seems unnecessary, https.get and https.request accept a URL object and [internally it makes use of urlToOptions](https://github.com/nodejs/node/blob/master/lib/https.js#L234), it makes sense to [leave node to handle the conversion](https://github.com/nodejs/node/blob/master/lib/internal/url.js#L1298) to a options object should anything change in future. 

The same applies to `ProxyAgent`, [it uses url.parse internally too](https://github.com/TooTallNate/node-https-proxy-agent/blob/master/index.js#L27). When I switched it to using the constructed object as you suggested above it broke.